File: solutions/python/2799-number-of-employees-who-met-the-target.py
class Solution:
    def countCompleteSubarrays(self, nums: List[int]) -> int:
        n = len(nums) 
        distinct_elements = len(set(nums)) 
        count = 0 
        left = 0 
        right = 0 
        counter = Counter() 

        while right < n: 
            counter[nums[right]] += 1 
            while len(counter) == distinct_elements: 
                counter[nums[left]] -= 1 
                if counter[nums[left]] == 0: 
                    del counter[nums[left]] 
                left += 1 
                count += n - right 
            right += 1 

        return count 
                                                                                                                                                                                                        

File: solutions/csharp/2799-number-of-employees-who-met-the-target.cs
public class Solution {
    public int CountCompleteSubarrays(int[] nums) {
        int n = nums.Length;
        HashSet<int> set = new HashSet<int>(nums);
        int distinct_elements = set.Count;
        int count = 0;
        int left = 0;
        int right = 0;
        Dictionary<int, int> counter = new Dictionary<int, int>();

        while (right < n) {
            if (!counter.ContainsKey(nums[right])) {
                counter[nums[right]] = 0;
            }
            counter[nums[right]]++;
            while (counter.Count == distinct_elements) {
                counter[nums[left]]--;
                if (counter[nums[left]] == 0) {
                    counter.Remove(nums[left]);
                }
                left++;
                count += n - right;
            }
            right++;
        }
        return count;        
    }
}
